content/en/docs/guides/onboarding-private-repo.md

```diff
@@ -0,0 +1,124 @@
+are onboarding, and visit [settings/tokens](https://github.com/settings/tokens) 
+to generate a Personal Access Token with `repo` scope. 
+2. Bootstrap your cluster with the token. Note that we add the `--token-auth` 
+flag, which lets Flux know that it should use your `GITHUB_TOKEN` permanently 
+for auth in the cluster.
+If you already bootstrapped your cluster when getting started but did not add 
+this flag, it is fine to re-run the bootstrap command. 
+```sh
+export GITHUB_TOKEN=$GITHUB_TOKEN
+flux bootstrap github \
+  --owner=$GITHUB_USER \
+  --repository=fleet-infra \
+  --branch=main \
+  --path=./clusters/my-cluster \
+  --token-auth
+```
+3.  You can verify that Flux has saved your `GITHUB_TOKEN` for ongoing 
+authentication by checking the `flux-system` secret in the `flux-system` 
+namespace:
+```sh
+kubectl get secret flux-system -n flux-system -o yaml
+# You should see the `data.username` and `data.password` fields
+```
+4. Now, when adding any new, private GitRepository, you can use the HTTP/S protocol in `spec.url`
+and reference the shared token `flux-system` Secret within `spec.secretRef`. This will authenticate
+the repository fetch without requiring an new, individual token or SSH deploy key to be generated.
+
+#### Example
+
+Create a [GitRepository](https://fluxcd.io/docs/components/source/gitrepositories/) source pointing to my-org/my-private-repo's master branch:
+```sh
+flux create source git podinfo \
+    --url=https://github.com/my-org/my-private-repo \
+    --branch=master \
+    --secret-ref=flux-system
+```
+
+The above command generates the following manifest:
+
+```yaml
+apiVersion: source.toolkit.fluxcd.io/v1beta1
+kind: GitRepository
+metadata:
+    name: my-private-repo
+    namespace: flux-system
+spec:
+    interval: 1m
+    url: https://github.com/my-org/my-private-repo
+    # since we are using a private GitHub repo, the below 
+    # section is needed to tell Flux to use the GitHub token in the 
+    # flux-system Secret for HTTP/S basic auth when syncing
+    secretRef:
+        name: flux-system
+    ref:
+        branch: master
+```
+
+### Note on namespaces
+
+**Note** that if you wish to manage your tenant `GitRepository` sources as part of a namespace that is _not_ flux-system (e.g. the `apps` namespace, [as in the multi-tenancy example](https://github.com/fluxcd/flux2-multi-tenancy/blob/main/tenants/base/dev-team/sync.yaml#L5)), the private source will not be able to read from the `flux-system` Secret.
+
+You will have to manually create a Kubernetes Secret with the `data.username` and `data.password` fields in the namespace that you are creating your `GitRepository` sources in. To keep Secrets private, consider using [kubeseal](sealed-secrets.md).
+```sh
+# for GitRepository private sources stored in the `apps` namespace
+kubectl create secret git repo-auth \
+    --namespace=apps \
+    --url=https://github.com/my-org/my-private-repo \
+	--username=$GITHUB_USER \
+	--password=$GITHUB_TOKEN
```

Comment:
This command creates the secret in-cluster, it needs `--export > my-secret.yaml`, I think it's best to link to https://fluxcd.io/docs/cmd/flux_create_secret_git/

Reply:
Thanks for pointing this out - I added the `--export` to the code example and linked to `flux create secret git`. At my org we don't check secrets into source control so we just created the Secret this way.